---
title: "Finish Aligning and Generate Read Counts"
author: "Pavan Narahari"
date: "2025-03-31"
output:
    rmdformats::robobook:
---

```{r setup, include=FALSE}
library(data.table)
library(tidyverse)
knitr::opts_chunk$set(echo = TRUE)
```

# Read Counts Table

```{r}
process_data_for_ggplot <- function(file, filter_categories) {
  summary_data <- read.table(file, header=TRUE, row.names = 1, sep = "\t")
  transposed_data <- as.data.frame(t(summary_data))
  transposed_data$Sample <- rownames(transposed_data)
  rownames(transposed_data) <- NULL
  transposed_data <- transposed_data %>% relocate(Sample)
  
  long_data <- transposed_data %>%
  pivot_longer(cols = -Sample, 
               names_to = "Status", 
               values_to = "Counts")
  
    sample_mapping <- c(
    "SRR21798749.SRR21798749.Aligned.sortedByCoord.out.bam" = "experimental_1",
    "SRR21798750.SRR21798750.Aligned.sortedByCoord.out.bam" = "experimental_2",
    "SRR21798751.SRR21798751.Aligned.sortedByCoord.out.bam" = "experimental_3",
    "SRR21798755.SRR21798755.Aligned.sortedByCoord.out.bam" = "control_1",
    "SRR21798756.SRR21798756.Aligned.sortedByCoord.out.bam" = "control_2",
    "SRR21798757.SRR21798757.Aligned.sortedByCoord.out.bam" = "control_3"
  )

  long_data$Sample <- recode(long_data$Sample, !!!sample_mapping)

  long_data <- long_data %>%
    filter(Status %in% filter_categories)

  return(long_data)
}
```

```{r}
gene_count_data = process_data_for_ggplot("musmusculus_fc.txt.summary", c("Assigned", "Unassigned_Ambiguity", "Unassigned_NoFeatures", "Unassigned_MultiMapping"))

head(gene_count_data)

plot <- ggplot(gene_count_data, aes(x = Counts, y = Sample, fill = Status)) +
  geom_bar(stat = "identity", position = "dodge") +
  labs(x = "# Reads", y = "Sample", title = "Reads for each Status Per Sample")

plot
```
